packages/ui/src/primitives/ChainIcon.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { FC } from 'react'
+import { reservoirChains } from '@reservoir0x/reservoir-sdk'
+import { ThemeContext } from '../ReservoirKitProvider'
+import { ReservoirKitThemeContext } from '../../stitches.config'
+import { useContext } from 'react'
+import MainnetIconDark from '../img/chains/MainnetIconDark'
+import MainnetIconLight from '../img/chains/MainnetIconLight'
+import PolygonIconDark from '../img/chains/PolygonIconDark'
+import PolygonIconLight from '../img/chains/PolygonIconLight'
+import BaseIconDark from '../img/chains/BaseIconDark'
+import BaseIconLight from '../img/chains/BaseIconLight'
+import Ancient8IconDark from '../img/chains/Ancient8IconDark'
+import Ancient8IconLight from '../img/chains/Ancient8IconLight'
+import ArbitrumIconDark from '../img/chains/ArbitrumIconDark'
+import ArbitrumIconLight from '../img/chains/ArbitrumIconLight'
+import ArbitrumNovaIconDark from '../img/chains/ArbitrumNovaIconDark'
+import ArbitrumNovaIconLight from '../img/chains/ArbitrumNovaIconLight'
+import AvalancheIconDark from '../img/chains/AvalancheIconDark'
+import AvalancheIconLight from '../img/chains/AvalancheIconLight'
+import BscIconDark from '../img/chains/BscIconDark'
+import BscIconLight from '../img/chains/BscIconLight'
+import GoerliIconDark from '../img/chains/GoerliIconDark'
+import GoerliIconLight from '../img/chains/GoerliIconLight'
+import LineaIconDark from '../img/chains/LineaIconDark'
+import LineaIconLight from '../img/chains/LineaIconLight'
+import OptimismIconDark from '../img/chains/OptimismIconDark'
+import OptimismIconLight from '../img/chains/OptimismIconLight'
+import ScrollTestnetIconDark from '../img/chains/ScrollTestnetIconDark'
+import ScrollTestnetIconLight from '../img/chains/ScrollTestnetIconLight'
+import PolygonZkevmIconDark from '../img/chains/PolygonZkevmIconDark'
+import PolygonZkevmIconLight from '../img/chains/PolygonZkevmIconLight'
+import ZksyncIconDark from '../img/chains/ZksyncIconDark'
+import ZksyncIconLight from '../img/chains/ZksyncIconLight'
+import ZoraIconDark from '../img/chains/ZoraIconDark'
+import ZoraIconLight from '../img/chains/ZoraIconLight'
+import { Box } from './'
+import { CSS } from '@stitches/react'
+
+type Props = {
+  chainId?: number
+  height?: number
+  css?: CSS
+}
+
+const ChainIcon: FC<Props> = ({ chainId, css = {}, height = 14 }) => {
+  const themeContext = useContext(ThemeContext)
+  const chainIcon: ReservoirKitThemeContext['assets']['chainIcon']['value'] =
+    themeContext && (themeContext as any)
+      ? themeContext['assets']['chainIcon']['value']
+      : 'dark'
+  let icon = null
+  switch (chainId) {
```

Comment:
feel like this should be a map but not a big deal

```js
const chainIcons = {
  [reservoirChains.ancient8Testnet.id]: {
    light: <Ancient8IconDark />,
    dark: <Ancient8IconLight />,
  }
  ...
}

maybe should even be part of the reservoir chain 👀

Reply:
I do like the idea of using a map. I think Pedro went with this structure so that there is always a fallback. We discussed attaching the chain icons to the reservoir chain, but decided against it because the reservoir chain object lives in the sdk and we this is soley for ui

Reply:
Updated to use a map called `chainIcons` within the constant folder